src/libcore/option.rs

```diff
@@ -837,9 +837,8 @@ impl<T> Option<T> {
     #[inline]
     #[stable(feature = "option_entry", since = "1.20.0")]
     pub fn get_or_insert_with<F: FnOnce() -> T>(&mut self, f: F) -> &mut T {
-        match *self {
-            None => *self = Some(f()),
-            _ => (),
+        if let None = *self {
+            *self = Some(f())
```

Comment:
nit: can you put a semicolon after the assignment?  I think of an elseless if as a statement, not a unit expression.

Reply:
I must have overlooked that, fixed.